src/Features/Core/Portable/ConvertAutoPropertyToFullProperty/AbstractConvertAutoPropertyToFullPropertyCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,199 @@
+        }
+
+        private async Task<Document> ExpandToFullPropertyAsync(
+            Document document,
+            SyntaxNode property,
+            IPropertySymbol propertySymbol,
+            SyntaxNode root,
+            CancellationToken cancellationToken)
+        {
+            var rules = await GetNamingRulesAsync(
+                document, 
+                cancellationToken).ConfigureAwait(false);
+            var fieldName = GenerateFieldName(propertySymbol, rules);
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+            var newRoot = await ExpandPropertyAndAddFieldAsync(
+                document,
+                property,
+                root,
+                propertySymbol,
+                fieldName,
+                generator,
+                cancellationToken).ConfigureAwait(false);
+
+            return document.WithSyntaxRoot(newRoot);
+        }
+
+        /// <summary>
+        /// Get the user-specified naming rules, then add standard default naming rules 
+        /// for both static and non-static fields.  The standard naming rules are added at the end 
+        /// so they will only be used if the user hasn't specified a preference.
+        /// </summary>
+        private static async Task<ImmutableArray<NamingRule>> GetNamingRulesAsync(
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            const string defaultStaticFieldPrefix = "s_";
+            const string defaultFieldPrefix = "_";
+
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var namingPreferencesOption = optionSet.GetOption(SimplificationOptions.NamingPreferences);
+            var rules = namingPreferencesOption.CreateRules().NamingRules
+                .AddRange(GetDefaultRule(ImmutableArray.Create(new ModifierKind(ModifierKindEnum.IsStatic)), defaultStaticFieldPrefix))
+                .AddRange(GetDefaultRule(ImmutableArray.Create<ModifierKind>(), defaultFieldPrefix));
+            return rules;
+        }
+
+        private static ImmutableArray<NamingRule> GetDefaultRule(
+            ImmutableArray<ModifierKind> modifiers,
+            string prefix)
+        {
+            return ImmutableArray.Create(
+                new NamingRule(
+                    new SymbolSpecification(
+                        Guid.NewGuid(),
+                        "Field",
+                        ImmutableArray.Create(new SymbolSpecification.SymbolKindOrTypeKind(SymbolKind.Field)),
+                        modifiers: modifiers),
+                    new NamingStyles.NamingStyle(
+                        Guid.NewGuid(),
+                        prefix: prefix,
+                        capitalizationScheme: Capitalization.CamelCase),
+                    DiagnosticSeverity.Hidden));
+        }
+
+        private string GenerateFieldName(IPropertySymbol property, ImmutableArray<NamingRule> rules)
```

Comment:
Should probably only be on the C# side, and then you won't make an unnecessary one for VB that just gets ignored later.